api/api.go

```diff
@@ -0,0 +1,65 @@
+// Package api defines implementations of endpoints and calls
+package api
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog"
+
+	"github.com/ovalfi/go-sdk/model"
+)
+
+// RemoteCalls abstracted definition of supported functions
+type RemoteCalls interface {
+	// Customer APIs
+	CreateCustomer(ctx context.Context, request model.CreateCustomerRequest) (model.Customer, error)
+	UpdateCustomer(ctx context.Context, request model.UpdateCustomerRequest) (model.Customer, error)
+	GetAllCustomers(ctx context.Context) ([]model.Customer, error)
+	GetCustomerByID(ctx context.Context, request model.GetCustomerByIDRequest) (model.CustomerInfo, error)
+
+	// Yield APIs
+	GetBusinessPortfolios(ctx context.Context) ([]model.Portfolio, error)
+	CreateYieldOfferingProfile(ctx context.Context, request model.CreateYieldOfferingProfilesRequest) (model.YieldOfferingProfile, error)
```

Comment:
I guess you wanna do the update and get yield offering profile in another PR

Reply:
Yes, I've done it already, but I don't want to push it to this PR

Reply:
Cool